src/coins.cpp

```diff
@@ -1025,6 +1025,35 @@ bool CCoinsViewCache::UpdateScInfo(const CScCertificate& cert, CBlockUndo& block
     return true;
 }
 
+bool CCoinsViewCache::IsOutputMature(const uint256& txHash, unsigned int pos) const
+{
+    if (HaveCoins(txHash) == 0)
```

Comment:
HaveCoins returns a boolean, consider replacing it with !HaveCoins()

Reply:
Agree, following bugfixing the HaveCoins + AccessCoins is changed to GetCoins which addresses both your points